motion/joystick_guidance/scripts/joystick_guidance.py

```diff
@@ -36,14 +132,106 @@ def callback(self, msg):
 		joystick_msg.torque.y = msg.axes[self.pitch]
 		joystick_msg.torque.z = msg.axes[self.yaw]
 
+		"""
+		# Uncomment if the thrust-vector should be given directly to the thrusters
 		self.pub.publish(joystick_msg)
+		"""
+
+		start_command = msg.buttons[self.start_button_idx]
+		if start_command == 1:
+			self.start_pressed = not self.start_pressed
+
+			bool_msg = Bool()
+			mool_msg.data = self.start_pressed
+			self.pub_state.publish(bool_msg)
+
+			# Sleep for 0.25 seconds to prevent the system from constantly switching between ROV and AUV
+			rospy.Duration(0.25).sleep()
+
+
+		# Calculating point and publishing to DP-controller
+		point = self.calculate_global_point(joystick_msg)
+
+		pose_msg = Pose()
+		pose_msg.position.x   	= point[0]
+		pose_msg.position.y   	= point[1]
+		pose_msg.position.z   	= point[2]
+		pose_msg.orientation.x 	= 0
+		pose_msg.orientation.y 	= 0
+		pose_msg.orientation.z 	= 0
+		pose_msg.orientation.w 	= 0
+
+		self.pub_joy.publish(pose_msg)
+
+	def calculate_global_point(self, joystick_msg):
+		"""
+		Calculates a point in the local frame based on the given thrust-vectors 
+		from joystick 
+
+		If required it must be converted into the global reference-frame
+		"""
+
+		# Scaling the force to get a linearized model
+		scaled_force_x, scaled_force_y, scaled_force_z = self.scale_force_vectors(joystick_msg)
+
+		local_calculated_point = [scaled_force_x, scaled_force_y, scaled_force_z]
+
+		# Calculating the length of the vectors
+		vector_length_square = 0
+		for i in range(len(local_calculated_point)):
+			vector_length_square += pow(local_calculated_point[i], 2)
+
+		# Normalizing if exceeding the max_point_range
+		if vector_length_square >= pow(self.max_point_range, 2):
+			# Over the set limit. Normalizing
+			vector_length = sqrt(vector_length_square)
+			local_calculated_point = [val / vector_length for val in local_calculated_point]
+
+		# Converting to the local point (Note that this will only work when we have inputs from /odometry/filtered)
+		global_calculated_point = convert_to_global(local_calculated_point, self.uuv_pose)
```

Comment:
global_calculated_point will currently fail if /odometry/filtered is not published to. It might be good to have an acceptance test for the uuv_pose message: if it is empty, make it some default pose.